app/src/features/surveys/observations/create/CreateObservationPage.tsx

```diff
@@ -0,0 +1,282 @@
+
+      const data = {
+        standardColumns: {
+          itis_scientific_name,
+          itis_tsn,
+          observation_date,
+          observation_time,
+          survey_sample_method_id,
+          survey_sample_period_id,
+          survey_sample_site_id,
+          latitude,
+          longitude,
+          count: observationPostObject.subcounts.reduce((sum, subcount) => sum + (subcount.subcount || 0), 0)
+        },
+        subcounts: observationPostObject.subcounts.map((subcount) => {
+          const { measurements, ...subcountProps } = subcount;
+
+          return {
+            observation_subcount_id: subcountProps.observation_subcount_id,
+            observation_subcount_sign_id: observationPostObject.standardColumns.observation_sign_id,
+            comment: subcountProps.comment,
+            subcount: subcountProps.subcount,
+            quantitative_measurements: measurements
+              .map((measurement) =>
+                'value' in measurement
+                  ? {
+                      measurement_id: measurement?.measurement_id,
+                      measurement_value: measurement?.value
+                    }
+                  : undefined
+              )
+              .filter((measurement) => measurement?.measurement_id !== undefined),
+            qualitative_measurements: measurements
+              .map((measurement) =>
+                'critterbase_measurement_qualitative_option_id' in measurement
+                  ? {
+                      measurement_id: measurement?.measurement_id,
+                      measurement_option_id: measurement?.critterbase_measurement_qualitative_option_id
+                    }
+                  : undefined
+              )
+              .filter((measurement) => measurement?.measurement_option_id !== undefined),
+            quantitative_environments: observationPostObject.standardColumns.environments
+              .map((environment) =>
+                'environment_quantitative_id' in environment
+                  ? {
+                      environment_quantitative_id: environment?.environment_quantitative_id,
+                      value: environment?.value
+                    }
+                  : undefined
+              )
+              .filter((environment) => environment?.environment_quantitative_id !== undefined),
+            qualitative_environments: observationPostObject.standardColumns.environments
+              .map((environment) =>
+                'environment_qualitative_id' in environment
+                  ? {
+                      environment_qualitative_id: environment.environment_qualitative_id,
+                      environment_qualitative_option_id: environment?.environment_qualitative_option_id
+                    }
+                  : undefined
+              )
+              .filter((environment) => environment?.environment_qualitative_id !== undefined)
+          };
+        })
+      } as ICreateObservationRequest;
```

Comment:
Rather than cast this to the type, we should just assign this to the const.

```
const data: ICreateObservationRequest = { ... }
```
Otherwise, there is no actual type safety. And actually if you make that change, the type isn't happy.

Partially related: It looks like the comment field is in the wrong spot. The form assigns it to the standard columns section, but its being added here under the subcounts section.

Reply:
fixed